docs/reference/analysis/tokenfilters/edgengram-tokenfilter.asciidoc

```diff
@@ -1,16 +1,193 @@
 [[analysis-edgengram-tokenfilter]]
-=== Edge NGram Token Filter
+=== Edge n-gram token filter
+++++
+<titleabbrev>Edge n-gram</titleabbrev>
+++++
 
-A token filter of type `edge_ngram`.
+Forms an https://en.wikipedia.org/wiki/N-gram[n-gram] of a specified length from
+the beginning of a token.
 
-The following are settings that can be set for a `edge_ngram` token
-filter type:
+For example, you can use the `edge_ngram` token filter to change `quick` to
+`qu`.
 
-[cols="<,<",options="header",]
-|======================================================
-|Setting |Description
-|`min_gram` |Defaults to `1`.
-|`max_gram` |Defaults to `2`.
-|`side` |deprecated. Either `front` or `back`. Defaults to `front`.
-|======================================================
+This filter uses Lucene's
+https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/ngram/EdgeNGramTokenFilter.html[EdgeNGramTokenFilter].
 
+[NOTE]
+====
+The `edge_ngram` filter is similar to the <<analysis-ngram-tokenizer,`ngram`
+token filter>>. However, the `edge_ngram` only outputs n-grams that start at the
+beginning of a token. These edge n-grams are useful for
+<<search-as-you-type,search-as-you-type>> queries.
+====
+
+[[analysis-edgengram-tokenfilter-analyze-ex]]
+==== Example
+
+The following <<indices-analyze,analyze API>> request uses the `edge_ngram`
+filter to convert `the quick brown fox jumps` to 1-character edge n-grams:
+
+[source,console]
+--------------------------------------------------
+GET _analyze
+{
+  "tokenizer": "standard",
+  "filter": [ "edge_ngram" ],
+  "text": "the quick brown fox jumps"
+}
+--------------------------------------------------
+
+The filter produces the following tokens:
+
+[source,text]
+--------------------------------------------------
+[ t, q, b, f, j ]
+--------------------------------------------------
+
```

Comment:
I'm a bit confused here, as the default settings are `min_gram` of 1 and `max_gram` of 2, which should surely produce `[ t, th, q, qu, b, br, f, fo, j, ju ]`?

Reply:
I experimented with this a bit more and found some odd behavior.

The following _analyze request produces only unigrams:

```
GET _analyze
{
  "tokenizer": "standard",
  "filter": [ "edge_ngram" ],
  "text": "the quick brown fox jumps"
}
```

However, treating `edge_ngram` as a custom filter with defaults produces both unigrams and bigrams:

```
GET _analyze
{
  "tokenizer": "standard",
  "filter": [
    {  "type": "edge_ngram" }
  ],
  "text": "the quick brown fox jumps"
}
```

I updated the analyze example to use the custom filter format with aeab02bb83724e819921b0b41ecb0e5678746ca1.

If you can, let me know if this is a bug, undocumented but expected behavior, or just my misunderstanding of how the `_analyze` API works. I'm happy to create a bug issue or document this behavior if needed.

Thanks!

Reply:
This looks like a discrepancy between the pre-configured token filter and the default settings for a custom filter; the pre-configured filter uses min & max gram of 1, but the custom defaults are 1 and 2.  It's a bit weird, but it's always been done like that, so I guess we just explicitly call it out in the documentation?

Reply:
Thanks @romseygeek. Added with dfd1e3b851ede061952a05d96973ab04c25c0b28.